src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -67,10 +67,10 @@ class ConsensusCallingIterator[ConsensusRead <: SimpleRead](sourceIterator: Iter
       groupingIterator.flatMap(caller.consensusReadsFromSamRecords)
     }
     else {
-      ParIterator(groupingIterator, threads=threads).flatMap { rs =>
+      ParIterator(groupingIterator, threads=threads, chunkSize=chunkSize).map { rs =>
         val caller = callers.get()
         caller.synchronized { caller.consensusReadsFromSamRecords(rs) }
-      }.toAsync(chunkSize * 8)
+      }.toAsync(chunkSize).flatten
     }
   }
```

Comment:
1. Before, `chunkSize` parameter was not actually passed down to the `ParIterator`, only the `toAsync` part. The default value of `1024` was hard-coded here, which is unexpected as it is an input member of the `ConsensusCallingIterator` class, and would be expected to be passed on.
2. My suggestion is to move the flattening of this mapping operation until after the `.toAsync` call. My reasoning here is that that, according to the documentation, this value should be a multiple of `chunkSize`. If we flatten before calling `toAsync`, we are buffering that number of records, rather than that number of tag families, which the input `ParIterator` is working on. 
  - For example, if your `chunkSize` is 10, and your family size is 100, and we assume two consensus reads per family, each chunk would contain 1000 reads. 
  - If we assume each family produces 2 consensus reads, then each chunk would produce 20 reads.
  - If we flatten before `.toAsync`, then we are using 8*20 reads, which is not a multiple of one chunk of input. In this case, the buffer would not even hold one chunk's worth of input reads in memory.